### PR TITLE
Adds inkyphat driver

### PR DESCRIPTION
**Pull request by blundell**
Makes an AndroidThings version of this python library: https://github.com/pimoroni/inky-phat

The InkyPhat is an e-ink display which we can draw white, black or red pixels.
The image stays on the screen even if there is no power to the device.

This PR allows:
- individual pixels to be drawn 
- images to be converted and drawn
- text to be drawn
- simple shapes to be drawn (circles, squares, triangles)

What it doesn't support:
- partial display updates (think like clipping draw areas)

### Pin Info

My Pi has the following config:
```
    private static final String INKY_PHAT_DISPLAY = "SPI0.0";
    private static final String COMMAND_PIN = "BCM22";
    private static final String RESET_PIN = "BCM27";
    private static final String BUSY_PIN = "BCM17";
```

### Sample usage

```java
import com.google.android.things.contrib.driver.inkyphat.InkyPhat;

// Access the display:

InkyPhat inkyPhat;

inkyPhat = InkyPhat.Factory.create(spiBusName, busyGpioPin, resetGpioPin, commanGpioPin, InkyPhat.Orientation.LANDSCAPE);

// Draw something:

Bitmap bitmap = BitmapFactory.decodeResource(resources, R.drawable.ic_logo);
inkyPhat.setImage(0, 0, bitmap, InkyPhat.Scale.FIT_X_OR_Y);

// Refresh the screen:

inkyPhat.refresh();

// Close the display when finished:

inkyPhat.close();
```

|InkyPhat Image Conversion|Original|
|----|----|
|image|logo|

|InkyPhat Image & Text|Original|
|----|----|
|image|ic_test_3|

|InkyPhat Image Portrait|Original|
|----|----|
|img-20170810-wa0013|ic_test_2|

|InkyPhat Image Landscape|Original|
|----|----|
|img-20170810-wa0021|img-20170810-wa0020|

|InkyPhat Icon conversion|Original|
|----|----|
|image|ic_test_1|

|InkyPhat Original Image|Original|
|----|----|
|img-20170810-wa0018|ic_test_4|

| Canvas Drawing |
|-----|
|webcam-toy-photo14|

### CODE REVIEW

File: inkyphat/src/main/java/com/google/android/things/contrib/driver/inkyphat/ColorConverter.java [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+import android.graphics.Color;
+
+class ColorConverter {
+
+    InkyPhat.Palette convertARBG888Color(int color) {
+        int red = Color.red(color);
+        int blue = Color.blue(color);
+        int green = Color.green(color);
+
+        if (red > 127 && blue > 127 && green > 127) {
```
Comment (xrigau) [inline inkyphat/src/main/java/com/google/android/things/contrib/driver/inkyphat/ColorConverter.java]: perhaps these should be configurable?

File: inkyphat/src/test/java/com/google/android/things/contrib/driver/inkyphat/InkyPhatTest.java [hunk 2]
```diff
@@ -0,0 +1,102 @@
...
+
+    @Test
+    public void busyIsInputPin() throws IOException {
+        verify(mBusyGpio).setDirection(Gpio.DIRECTION_IN);
+    }
+
+    @Test
+    public void close_closesAllPeripherals() throws IOException {
+        inkyPhat.close();
+        Mockito.verify(mSpi).close();
```
Comment (xrigau) [inline inkyphat/src/test/java/com/google/android/things/contrib/driver/inkyphat/InkyPhatTest.java]: I'd change this to use the static import for consistency

**Comment by proppy** [comment]
Apparently there is a new hardware version: http://blog.pimoroni.com/the-murky-depths-of-inky-phat/ !

**Comment by blundell** [comment]
awesome! I'll get the new display ordered and test this is still working

> The end result is that the library should function identically whether you're using either display and automagically detect the display type and use the correct protocol for that display.

In theory it should ;-) but we'll see

**Comment by blundell** [comment]
Tested compatibility and had to add a new driver for version 2. It's not complete but I pushed what I have in case people want to help out. Will add screenshots of v2 hardware when complete.

**Comment by Fleker** [comment]
Good morning,
We've recently [added a program](https://androidthings.withgoogle.com/submit-a-driver/#!/) to let developers submit drivers from their own repository to androidthings.withgoogle.com. Can you put this change in your own repository and then submit through that site?

**Pull request closed without merging**